### PR TITLE
Migrate specs update to updatecli

### PR DESCRIPTION
**Pull request by reakaleek**
## What

Migrate https://github.com/elastic/apm/blob/main/.ci/Jenkinsfile to the consumer using updatecli and GH Actions workflows.

## Why

Jenkins 2 GH Actions Migration

## How to test

```shell
export GITHUB_TOKEN="<your-personal-access-token>"
export GIT_USER="FooBar"
export GIT_EMAIL="foo@bar.com"
updatecli --config .ci/updatecli.d/update-json-specs.yml diff
updatecli --config .ci/updatecli.d/update-gherkin-specs.yml diff
```

### CODE REVIEW

File: .ci/updatecli.d/update-json-specs.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    kind: file
+    spec:
+      file: https://raw.githubusercontent.com/elastic/apm/main/tests/agents/json-specs/w3c_distributed_tracing.json
+  wildcard_matcher_tests.json:
+    kind: file
+    spec:
+      file: https://raw.githubusercontent.com/elastic/apm/main/tests/agents/json-specs/wildcard_matcher_tests.json
+actions:
+  pr:
+    title: Update JSON specs
```
Comment (v1v) [inline .ci/updatecli.d/update-json-specs.yml:56]: title

Comment (reakaleek) [reply .ci/updatecli.d/update-json-specs.yml]: 😅 f5f0fe7

File: .ci/updatecli.d/update-gherkin-specs.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
+name: update-gherkin-specs
+pipelineid: update-gherkin-specs
+
```
Comment (v1v) [inline .ci/updatecli.d/update-gherkin-specs.yml:3]:
```suggestion
name: update-gherkin-specs
pipelineid: update-gherkin-specs
title: synchronize gherkin specs

```

File: .ci/updatecli.d/update-gherkin-specs.yml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    kind: file
+    spec:
+      file: https://raw.githubusercontent.com/elastic/apm/main/tests/agents/gherkin-specs/user_agent.feature
+
+actions:
+  pr:
+    kind: "github/pullrequest"
+    scmid: default
+    spec:
+      title: Update Gherkin specs
```
Comment (v1v) [inline .ci/updatecli.d/update-gherkin-specs.yml:56]:
```suggestion
    title: '[Automation] Update Gherkin specs'
    spec:
```

File: .ci/updatecli.d/update-json-specs.yml [hunk 4]
```diff
@@ -0,0 +1,120 @@
+name: update-jsons-specs
+pipelineid: update-json-specs
```
Comment (v1v) [inline .ci/updatecli.d/update-json-specs.yml:2]:
```suggestion
name: update-json-specs
pipelineid: update-json-specs
title: synchronize json specs
```

**Pull request merged**